### DISCUSSION TITLE
Access auth state from the backend

### DISCUSSION THREAD

**Discussion by H1Mohamed** [pocketbase/pocketbase#5190] [Q&A]
I have a sveltekit app where both ssr and csr are enabled, I want to access the user auth state from the server before the webpage's code is sent to the client,is it possible to achieve it in the javascript sdk?

**Comment by rmoskal** [comment]
The documentation shows how:

https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state

However at least with version 0.22.16  running inside of a docker container, these values are always null.

I'd greatly appreciate it if someone could verify that this is working.

**Reply by ganigeorgiev** [reply]
The above works only when using the JS pb_hooks which doesn't seems to be what the author is asking.

> However at least with version 0.22.16 running inside of a docker container, these values are always null.

If it is `null` then you are not actually authenticated (eg. you are not proxieing properly the `Authorization` header to your Docker container). But again, this is not related to the above question (which I don't fully understand to provide an answer to). If you are still not able to identify the culprit, please open a new Q&A discussion elaborating more on your problem, ideally with some some steps to reproduce.

**Comment by H1Mohamed** [comment]
it seems I didn't clarify my question correctly, I want to know if the user is logged in from the server using a load function before serving them the website,is it possible?

**Reply by ganigeorgiev** [reply]
Unfortunately the above still doesn't clarify for me what actually you are trying to do, nor how do you initialize the JS SDK in your mixed SvelteKit SSR-CSR setup.

In general if you want a mixed access to the JS SDK (aka. both in the browser and node), then you'll need to maintain 2 different JS SDK instances (_for the client it is OK to be global, but in the JS SSR you need a new instance for each request_).
The state between the 2 instances can be shared via a cookie that is exported with `httpOnly: false`. Similar to the [SSR integration examples](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#ssr-integration) for the client instance you need to update the cookie but instead of modifying the `response` object (there is no such in the browser) you need to update `document.cookie`. An example browser client could be:

```js
import PocketBase from "pocketbase"

const pb = new PocketBase("...")

pb.authStore.loadFromCookie(document.cookie)

pb.authStore.onChange(() => {
  document.cookie = pb.authStore.exportToCookie({ httpOnly: false }); // "httpOnly: false" for consistency with the node server
});

export default pb
```

With that said, if you are new to web development  I really don't recommend using JS SSR with PocketBase because you most likely will mess up the security. With the refactoring I'll update the documentation with a more detailed note about this but for now you can find some ramblings in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032, https://github.com/pocketbase/pocketbase/discussions/4962, https://github.com/pocketbase/pocketbase/discussions/4412#discussioncomment-8575834 and probably others.
